src/store.rs

```diff
@@ -1023,22 +752,32 @@ impl<F: LurkField> Store<F> {
         }
     }
 
-    pub fn fetch(&self, ptr: &Ptr<F>) -> Option<Expression<'_, F>> {
+    pub fn fetch(&self, ptr: &Ptr<F>) -> Option<Expression<F>> {
         if ptr.is_opaque() {
-            return Some(Expression::Opaque(*ptr));
+            return None;
         }
         match ptr.tag {
             ExprTag::Nil => Some(Expression::Nil),
             ExprTag::Cons => self.fetch_cons(ptr).map(|(a, b)| Expression::Cons(*a, *b)),
             ExprTag::Comm => self.fetch_comm(ptr).map(|(a, b)| Expression::Comm(a.0, *b)),
-            ExprTag::Sym => self.fetch_sym(ptr).map(|sym| Expression::Sym(sym)),
-            ExprTag::Key => self.fetch_sym(ptr).map(|sym| Expression::Sym(sym)),
+            ExprTag::Sym if ptr.raw.is_null() => Some(Expression::SymNil),
+            ExprTag::Sym => self
+                .fetch_symcons(ptr)
+                .map(|(car, cdr)| Expression::SymCons(car, cdr)),
+            ExprTag::Key => Some(Expression::Key(Ptr {
+                tag: ExprTag::Sym,
```

Comment:
Do we really need to do this? If we're abusing the tag like this, can we maybe just leave it out of the store?